.pre-commit-config.yaml

```diff
@@ -47,3 +47,15 @@ repos:
         language: rust
         files: \.rs$
         pass_filenames: false
+      - id: conventional-commit-msg-validation
+        name: commit message conventional validation
+        language: pygrep
+        entry: '^(breaking|build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test){1}(\([\w\-\.]+\))?(!)?: ([\w ])+([\s\S]*)'
```

Comment:
The `([\w ])+` seems only match word characters.

Reply:
The `\w` in regular expressions represents any word characters. It matches any alphanumeric character, like a-z, A-Z, 0-9, and underscore. It's equivalent to the character class`[a-zA-z0-9_]`

Reply:
for example it will not match
```
feat: `foo`
```
maybe `^(breaking|build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test){1}(\([\w\-\.]+\))?(!)?: .+([\s\S]*)` is better?

Reply:
IMO, using the wildcard "." is not a good choice. While it allows you to use backticks, you can also use other symbols as well, like #$%^ etc. How about "^(breaking|build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test){1}(\([\w\-\.]+\))?(!)?: ([\w `])+([\s\S]*)"?

Reply:
makes sense